src/XMakeBuildEngine/Construction/ProjectElementContainer.cs

```diff
@@ -170,6 +172,18 @@ public void InsertAfterChild(ProjectElement child, ProjectElement reference)
             }
 
             XmlElement.InsertAfter(child.XmlElement, reference.XmlElement);
+            if (XmlDocument.PreserveWhitespace)
+            {
+                //  If we are trying to preserve formatting of the file, then the new node won't automatically be indented.
+                //  So try to match the surrounding formatting by checking the whitespace that precedes the node we inserted
+                //  after, and inserting the same whitespace between the previous node and the one we added
+                if (reference.XmlElement.PreviousSibling != null &&
+                    reference.XmlElement.PreviousSibling.NodeType == XmlNodeType.Whitespace)
+                {
+                    var newWhitespaceNode = XmlDocument.CreateWhitespace(reference.XmlElement.PreviousSibling.Value);
+                    XmlElement.InsertAfter(newWhitespaceNode, reference.XmlElement);
+                }
```

Comment:
Yeah, I didn't see an elegant way to unify the different cases that exist within `ProjectElementContainer`.